### PR TITLE
Improve hand scene by splitting different API examples

### PR DESCRIPTION
**Pull request by keveleigh**
image

Splits the MR OpenXR extension API example to be distinctly toggleable from the Unity feature usage example.
Also disables the MRTK hand visualization upon entering the scene, to defer to the scene representations without visual collision.

### CODE REVIEW

**Review by KBonds — APPROVED** [review]
Just tried this out, it looks great! I'll update the readme in #5 to reflect the changes here

File: Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            UpdateHandJoints(HandTracker.Left, leftHand, FrameTime.OnUpdate);
+            UpdateHandJoints(HandTracker.Right, rightHand, FrameTime.OnUpdate);
+        }
+
+        private static void UpdateHandJoints(HandTracker handTracker, Hand hand, FrameTime frameTime)
+        {
+            if (handTracker.TryLocateHandJoints(frameTime, HandJointLocations))
+            {
+                hand?.UpdateHandJoints(HandJointLocations);
```
Comment (yl-msft) [inline Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:52]: the "hand.cs" script still have mixed logic of feature usages and openxr extensions.  Suggest to split them into these classes as private classes.  One indicator is the "using namespaces" above are clean without the other API set in all C# scripts.

File: Assets/HandTracking/Scripts/ToggleMRTKHandVisualization.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit;
+using Microsoft.MixedReality.Toolkit.Input;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.OpenXR.Samples
+{
+    /// <summary>
+    /// Turns the MRTK hand visualization off when the scene is loaded, to defer to the sample scene's representation.
```
Comment (yl-msft) [inline Assets/HandTracking/Scripts/ToggleMRTKHandVisualization.cs:11]: Love this comment.  This sample itself is already a very good example of how to use MRTK.  I will remember this.

Can you help add more comments to other sample code as well when you get there?  Thanks

File: Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs (hunk 1 above)
Comment (keveleigh) [reply Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:53]: Split them out!

File: Assets/HandTracking/Scripts/FeatureUsageHandJointsManager.cs [hunk 3]
```diff
@@ -56,5 +56,154 @@ private static void UpdateHandJoints(InputDeviceCharacteristics flag, Hand hand)
                 hand?.DisableHandJoints();
             }
         }
+
+        private class Hand
+        {
+            public Hand(GameObject handJointPrefab)
+            {
+                this.handJointPrefab = handJointPrefab;
+                handRoot = new GameObject("HandParent");
```
Comment (KBonds) [inline Assets/HandTracking/Scripts/FeatureUsageHandJointsManager.cs:65]: Should we destroy these GameObjects? I think I've seen these accumulate over time, especially in some scenarios where a scene stays loaded for the duration of the app lifetime while others are loaded additively.

Comment (keveleigh) [reply Assets/HandTracking/Scripts/FeatureUsageHandJointsManager.cs:65]: Oh good call!

Comment (keveleigh) [reply Assets/HandTracking/Scripts/FeatureUsageHandJointsManager.cs:71]: Added cleanup code

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+                // Lighten the colors as we go up the finger.
+                int i = index ?? 0;
+                float saturation = 1.0f - (i * 0.15f);
+                Color.RGBToHSV(jointColor, out float h, out float _, out float v);
+                jointObject.GetComponent<Renderer>().material.color = Color.HSVToRGB(h, saturation, v);
+            }
+
+            /// <summary>
+            /// These indices map to the order they're provided through Unity's HandData.
```
Comment (yl-msft) [inline BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:196]: this comment is not correct.  It is indices to map to the coloring of joints, right?

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+                const string Thumb = "Thumb";
+                const string Index = "Index";
+                const string Middle = "Middle";
+                const string Ring = "Ring";
+                const string Little = "Little";
+
+                if (fingerName.StartsWith(Thumb))
+                {
+                    jointColor = (index == 0) ? Color.magenta : Color.red;
```
Comment (yl-msft) [inline BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:169]: Is this still necessary?

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+                    gameObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                    Destroy(gameObject.GetComponent<Collider>());
+                    gameObject.transform.localScale = new Vector3(0.015f, 0.015f, 0.015f);
+                    gameObject.transform.parent = handRoot.transform;
+                    return true;
+                }
+            }
+
+            /// <summary>
+            /// Update this hand's internal state with the Unity XR InputDevice's hand data.
```
Comment (yl-msft) [inline BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:105]: this comment is out dated.

**Review by yl-msft — APPROVED** [review]

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs (hunk 5 above)
Comment (keveleigh) [reply BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs:169]: Good call! It's not

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs (hunk 4 above)
Comment (keveleigh) [reply BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs]: Updated!

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs (hunk 5 above)
Comment (keveleigh) [reply BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs]: Updated!

File: BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs (hunk 6 above)
Comment (keveleigh) [reply BasicSample/Assets/HandTracking/Scripts/OpenXRExtensionHandJointsManager.cs]: Updated!

**Pull request merged**